File: UVa/Contents/a519/a519.c
#include<stdio.h>

void fibTable(void);

long long fib[81];

int main(void)
{
	int n;
	
	fibTable();
	
	while(1)
	{
		scanf("%d", &n);
		
		if(!n)
			break;
		
		printf("%lld\n", fib[n]);
	}
	
	return 0;
}

void fibTable()
{
	fib[0] = 1;
	fib[1] = 1;
	
	for(int i = 2; i < 81; i++)
		fib[i] = fib[i - 1] + fib[i - 2];
}

File: UVa/Contents/a519/a519.py
import sys

fib = [0] * 81
def fibTable():
    fib[0] = fib[1] = 1
    for i in range(2, 81):
        fib[i] = fib[i - 1] + fib[i - 2]

for n in sys.stdin:
    n = int(n)

    if n == 0:
        break

    fibTable()
    print(fib[n])
